sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashContainerChangeFeedTest.java

```diff
@@ -0,0 +1,835 @@
+        }
+
+        // applying updates
+        updateAction.run();
+
+        for (int i = 0; i < 20; i++) {
+            String pkValue = partitionKeyToDocuments.keySet().stream().skip(i).findFirst().get();
+            logger.info(String.format("Validation after updates - PK value: '%s'", pkValue));
+
+            final int expectedEventCountAfterUpdates = i < 5 ?
+                i < 1 ? 0 : 2  // on the first logical partitions all updated documents were deleted
+                : 0; // no updates
+
+            CosmosChangeFeedRequestOptions options = CosmosChangeFeedRequestOptions
+                .createForProcessingFromContinuation(continuations.get(pkValue));
+
+            drainAndValidateChangeFeedResults(options, null, expectedEventCountAfterUpdates);
+        }
+    }
+
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void asyncChangeFeed_fromBeginning_incremental_forFeedRange() throws Exception {
+        this.createContainer(
+            (cp) -> cp.setChangeFeedPolicy(ChangeFeedPolicy.createLatestVersionPolicy())
+        );
+        insertDocuments(200, 7);
+        updateDocuments(3, 5);
+        deleteDocuments(2, 3);
+        Runnable updateAction = () -> {
+            updateDocuments(5, 2);
+            deleteDocuments(1, 3);
+        };
+
+        final int expectedTotalInitialEventCount =
+            200 * 7   //inserted
+                + 0       // updates won't show up as extra events in incremental mode
+                - 2 * 3;  // updated then deleted documents won't show up at all in incremental mode
+
+        final int expectedTotalEventCountAfterUpdates =
+            5 * 2     // event count for initial updates
+                - 1 * Math.min(2,3);   // reducing events for 2 of the 3 deleted documents
+        // (because they have also had been updated)
+
+        List<FeedRange> feedRanges = createdContainer.getFeedRanges();
+        List<CosmosChangeFeedRequestOptions> options = new ArrayList<>();
+
+        for (int i = 0; i < feedRanges.size(); i++) {
+            options.add(CosmosChangeFeedRequestOptions
+                .createForProcessingFromBeginning(feedRanges.get(i)));
+        }
+
+        final Map<Integer, String> continuations = drainAndValidateChangeFeedResults(
+            options,
+            null,
+            expectedTotalInitialEventCount);
+
+        // applying updates
+        updateAction.run();
+
+        options.clear();
+        for (int i = 0; i < feedRanges.size(); i++) {
+            options.add(CosmosChangeFeedRequestOptions
+                .createForProcessingFromContinuation(continuations.get(i)));
+        }
+
+        drainAndValidateChangeFeedResults(options, null, expectedTotalEventCountAfterUpdates);
+    }
+
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void asyncChangeFeed_fromBeginning_fullFidelity_forFullRange() throws Exception {
+        assertThrows(
```

Comment:
We probably don't need this test, as it doesn't checks anything related to subpartitioning, right?